src/components/transport_manager/src/usb/libusb/usb_connection.cc

```diff
@@ -81,6 +81,7 @@ UsbConnection::~UsbConnection() {
   LOG4CXX_TRACE(logger_, "enter with this" << this);
   Finalise();
   libusb_free_transfer(in_transfer_);
+  libusb_free_transfer(out_transfer_);
```

Comment:
@dev-gh do we need to free out transfer here? Seems like it will be freed in `UsbConnection::OnOutTransfer()` callaback.

Reply:
@AKalinich-Luxoft 
```
if (!current_out_message_.valid()) {
    libusb_free_transfer(transfer);
    out_transfer_ = NULL;
    waiting_out_transfer_cancel_ = false;
  }
```
If you mean this code than I think its being done only in specific case - when message is invalid.

Reply:
@dev-gh in this specific case out_transfer will be a NULL. Is that OK to call `libusb_free_transfer` with NULL pointer in destructor?

Reply:
@AKalinich-Luxoft 
Yep
```
/** \ingroup asyncio
 * Free a transfer structure. This should be called for all transfers
 * allocated with libusb_alloc_transfer().
 *
 * If the \ref libusb_transfer_flags::LIBUSB_TRANSFER_FREE_BUFFER
 * "LIBUSB_TRANSFER_FREE_BUFFER" flag is set and the transfer buffer is
 * non-NULL, this function will also free the transfer buffer using the
 * standard system memory allocator (e.g. free()).
 *

 * It is legal to call this function with a NULL transfer. In this case,
 * the function will simply return safely.
 *

 * It is not legal to free an active transfer (one which has been submitted
 * and has not yet completed).
 *
 * \param transfer the transfer to free
 */
void API_EXPORTED libusb_free_transfer(struct libusb_transfer *transfer)
```